.github/workflows/build-test.yml

```diff
@@ -9,119 +9,105 @@ on:
   pull_request:
     branches: [ master, release* ]
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   pre-flight-check:
-     runs-on: ubuntu-20.04
-     steps:
-       - uses: actions/checkout@v2
-       - run: ./tools/check_formatting.sh
-
-  # This job builds & runs iceoryx tests in ubuntu-18.04
-  iceoryx-ubuntu:
-     runs-on: ubuntu-18.04
-     needs: pre-flight-check
-
-     # Steps represent a sequence of tasks that will be executed as part of the job
-     steps:
-       - name: compiler versions
-         run: |
-          echo $(gcc --version)
-          echo $(clang --version)
-
-       - name: Install iceoryx dependencies
-         # Softwares installed in ubuntu-18.04 instance
-         # https://github.com/actions/virtual-environments/blob/master/images/linux/Ubuntu1804-README.md
-         run: |
-          sudo apt-get update
-          sudo apt-get install -y libacl1-dev libncurses5-dev bison
-
-       - name: Checkout
-         uses: actions/checkout@v2
-
-       - name: Build sources
-         run: |
-            sudo $GITHUB_WORKSPACE/tools/add_test_users.sh
-            $GITHUB_WORKSPACE/tools/iceoryx_build_test.sh build-strict build-all out-of-tree build-shared test-add-user
-
-       - name: Build debian package
-         run: |
-            $GITHUB_WORKSPACE/tools/iceoryx_build_test.sh package
-
-       - name: Run all tests
-         run: |
-           cd $GITHUB_WORKSPACE/build
-           tools/run_tests.sh all
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - run: ./tools/check_formatting.sh
 
-       - name: Build RouDi examples without TOML support
-         run: |
-           $GITHUB_WORKSPACE/tools/iceoryx_build_test.sh out-of-tree examples toml-config-off clean
+  build-test-ubuntu:
+    runs-on: ubuntu-18.04
+    needs: pre-flight-check
+    steps:
+      - uses: actions/checkout@v2
+      - run: ./tools/ci/build-test-ubuntu.sh
 
- # This job builds & runs iceoryx tests in macos-10.14
-  iceoryx-macos:
+  build-test-macos:
     runs-on: macos-10.14
-    # Softwares installed in macos-latest instance
-    # https://github.com/actions/virtual-environments/blob/master/images/macos/macos-10.14-Readme.md
+    # prevent stuck jobs consuming macos runners for 6 hours
+    timeout-minutes: 30
```

Comment:
@dkroenke I see macos jobs can stuck for 6 hours (see [here](https://github.com/eclipse-iceoryx/iceoryx/runs/3242681619)), so we should upper-bound them.

There are seems multiple issues:
1. real root cause that should be investigated (happened on master, nothing to do with this change):
```
Warning:  /Users/runner/work/iceoryx/iceoryx/build/dependencies/googletest/src/googletest/src/gtest-death-test.cc:1050:: Death tests use fork(), which is unsafe particularly in a threaded context. For this test, Google Test detected 6 threads. See https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#death-tests-and-threads for more explanation and suggested solutions, especially if this is the last message you see before your test times out.
Error: The operation was canceled.
```

2. the case, when there are no runners available, I was hit today twice, when step was not even scheduled to run for a while, timeout should mitigate this second issue at least

Reply:
@dkroenke I created issue https://github.com/eclipse-iceoryx/iceoryx/issues/898